### ISSUE TITLE
 socket:recv() / socket:recvfrom() updates

### ISSUE THREAD

**Issue by TarantoolBot** [tarantool/doc#606]
- API change (for a UDP socket after #3619): a datagram will not be truncated to 512 bytes and the input buffer size will be evaluated when no explicit size was passed (but the evaluating leads to extra syscall); check the commit message for more information.
- API undocumented (for a UDP socket): a datagram will be truncated to `size` length in case when explicit size was passed, but the datagram length is larger then the size; the tail of the datagram will be discarded in the case; don't forget to note TCP behaviour (it is very different); check the commit message for more information.
- Recommendation missed: recommended to set size explicitly for recv and recvfrom; don't forget to give recommendations whether it is possible to determine the size (message format, network conditions) and how to do so depending of network conditions (add note that datagram sizes above MTU will lead to IP fragmentation that can be inefficient; add note that there is tuneable `sysctl net.inet.udp.maxdgram` on Mac OS); check the commit message for more information.
- Minor: update 'Use of a UDP socket on localhost' to use explicit size (say, 512), because we will recommend it for readers.
- Minor: ToC sorting is such that `recv` is far away from `recvfrom`.
- Monor: inconsistency in parameters naming: `recv(size)` vs `recvfrom(limit)`. The `size` term should be preferred IMHO, because AFAIK the word 'size' usually applied for the size of a buffer (while 'length' usually applied to a message length).

Feel free to ask more informatiom from @Totktonada.
Requested by @Totktonada in https://github.com/tarantool/tarantool/issues/3619.

**Comment by Totktonada** [comment]
@lenkis Should I file the separate issue re ToC sorting?

**Comment by pgulutzan** [comment]
@lenkis: I didn't move recvfrom to be with recv, because currently it is with sendto, and the sendto/recvfrom relationship looked to me just as logical as the recv/recvfrom relationship. It was described as "minor".

**Comment by Totktonada** [comment]
@pgulutzan Got it. Maybe place all them three near?

**Comment by pgulutzan** [comment]
@Toktonada, @lenkis: We can put any or all of the "Module socket" functions in a different order. Just say what specifically you want moved and where, no problem.

**Comment by Totktonada** [comment]
It is hard to propose strict sorting/grouping criteria and it is really minor thing. Nevermind.
